packages/ui/src/components/branch-selector.tsx

```diff
@@ -0,0 +1,70 @@
+import {
+  Button,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+  Icon,
+  Text
+} from '@/components'
+import { cn } from '@utils/cn'
+
+export interface BranchListProps {
+  name?: string
+}
+
+interface PageProps {
+  name: string
+  branchList?: BranchListProps[]
+  size?: 'default' | 'sm'
+  width?: 'auto' | 'sm' | 'md' | 'lg' | 'full'
+  selectBranch: (branch: string) => void
+  prefix?: string
+}
+
+export const BranchSelector = ({ ...props }: PageProps) => {
+  const { name, branchList, size = 'default', selectBranch, width = 'auto', prefix = undefined } = props
+
+  const widthClasses: { [key in NonNullable<PageProps['width']>]: string } = {
+    auto: 'w-auto',
+    sm: 'w-16',
+    md: 'w-32',
+    lg: 'w-48',
+    full: 'w-full'
+  }
+
+  return (
+    <DropdownMenu>
+      <DropdownMenuTrigger asChild>
+        <Button
+          variant="secondary"
+          size={size}
+          className={cn(widthClasses[width], 'flex items-center gap-1.5 overflow-hidden px-3', {
+            'bg-muted': !prefix,
+            'bg-background': prefix,
+            'border-border': prefix,
+            'border-2': prefix
+          })}
+        >
+          {prefix ? null : <Icon name="branch" size={12} className="text-tertiary-background min-w-[12px]" />}
```

Comment:
Is there an official spacing unit we can use instead of a specific pixel value of 12?